File: erlang/pe001.erl
-module(pe001).
-export([start/0]).

max() -> 999.

start() ->
  pe001(max(), 0).
  
pe001(N, Result) when N > 0, N rem 3 == 0; N rem 5 == 0 ->
    pe001(N-1, N+Result);
    
pe001(N, Result) when N > 0 ->
    pe001(N-1, Result);
    
pe001(N, Result) ->
    io:format("~w~n", [Result]).

File: elixir/pe001.exs
defmodule ProjectEuler do
  def pe001(n) do
    sum_multiples(n, 0)
  end
   
  defp sum_multiples(n, sum) when n > 0 and rem(n, 3) == 0 or rem(n, 5) == 0 do
    sum_multiples(n - 1, sum + n)
  end

  defp sum_multiples(n, sum) when n > 0 do
    sum_multiples(n - 1, sum)
  end

  defp sum_multiples(_, sum) do
    sum
  end
end

IO.puts(ProjectEuler.pe001(999))
